src/test/compile-fail/issue-29859.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(optin_builtin_traits)]
+
+trait Magic: Copy {} //~ ERROR E0565
```

Comment:
Can you add a comment explaining the purpose of this test?

Also, I think I'd prefer the name `typeck-default-trait-no-supertraits.rs` (and reference issue #29859 in the comment). 

(Probably best to rename the other test too.)